packages/front-end/components/Settings/Team/SingleRoleSelector.tsx

```diff
@@ -33,10 +34,42 @@ export default function SingleRoleSelector({
     roleOptions = roles.filter((r) => r.id !== "noaccess");
   }
 
+  if (!includeAdminRole) {
+    roleOptions = roles.filter((r) => r.id !== "admin");
```

Comment:
This should be `roleOptions.filter`.  Otherwise, it will overwrite the above `isNoAccessRoleEnabled` logic.